charts/replicated-library/values.yaml

```diff
@@ -116,6 +116,7 @@ apps:
     initContainers: {}
 
     # -- Specify a list of volumes that get mounted to the app.
+    # persistentVolumeClaims which are present and enabled in the persistence configuraiton will have the prefix added automatically.
```

Comment:
I like the spirit of this, but I feel like we'd need to be consistent with everything in the chart that can be cross-referenced or else this would be confusing no? Environment variables, configmaps, secrets, etc.

Reply:
Yup I agree, either we do this for everything or we don't try and do it at all. I'm hoping the logic to do this is fairly reusable so it's not a big lift to port this onto all the objects we cross references. 

If we _don't_ do this then doing things you can never map volumes, secrets, configs from the values.yaml and you have to always go directly into the `.tpl` files. Maybe that's fine, but that's what I was trying to avoid.

Reply:
I think the implementation here makes sense, but I think the discovery in where to use the prefix for environment variables, ConfigMaps, etc will be the difficult part. Similar to what i'm facing in the ConfigMap/Secret sha annotation work. 

Maybe we should just document:
1. How to template to cross reference objects
2. How to override the prefix and set it to "", so you can get a consistent name. I feel more apps than not don't actually need to support multiple deploys to the same namespaces especially for those using Replicated.

Reply:
The more I look at it the more I'm coming around to that opinion as well. I really like the idea of templating these for the user, but I do fear the complexity of handling it generically in the Library isn't enough value over just having a chart author template their specific needs.

Let me think about it some more, I'm really concerned that "charts can never be deployed twice" is such a big restriction I'm hesitant to make that call. I'm not aware of any other chart that I use with that restriction...

Reply:
Ok we just synced on this and here's what we're thinking is a good compromise.

As Diamon was saying above.
1. Set the fullNameOverride in the global defaults to something which turns it off by default. This lets people just 'do the easy thing' and hardcode their names.
2. Document how to do prefix templating if someone does want and need to turn it on.
3. Keep whatever logic we do have and make an attempt to handle it where we can in the chart, but recognize we can never fully close the gap so this is intentionally an advanced feature that will have some holes. For example, there is no clear way for us to automatically handle if you pass a service name in an `env` on your container. The chart author would just need to do their own templating as documented in 2. above.

Reply:
@diamonwiggins with that, are we good to merge this one and we'll do the global default change in it's own PR?